Annotato/Annotato/ViewModels/Document/DocumentAnnotationViewModel.swift

```diff
@@ -29,4 +46,31 @@ class DocumentAnnotationViewModel {
     var startingAnnotationType: AnnotationType? {
         isNew ? self.parts[0].annotationType : nil
     }
+
+    func updateLocation(documentViewPoint: CGPoint, parentView: UIView?) {
+        guard let parentView = parentView else {
+            return
+        }
+        guard let pdfView = parentView.superview?.superview
+        as? DocumentPdfView else {
+            return
+        }
+
+        // Get coordinates in terms of the visible view page space
+        let visibleViewCoordinates = parentView.convert(
+            documentViewPoint, to: parentView.superview?.superview)
+
+        // Get the current page that this annotation is at
+        guard let currPage = pdfView.page(
+            for: visibleViewCoordinates, nearest: true) else {
+            return
+        }
+        // Get the page space coordinates
+        let pageSpaceCoordinates = pdfView.convert(visibleViewCoordinates, to: currPage)
```

Comment:
Seems like the reason why parent view is required, is to calculate `pageSpaceCoordinates`. Can this calculation be done within the view layer, and we simply pass the coordinates to this method?
```
func updateLocation(documentViewPoint: CGPoint, pageSpaceCoordinates: CGPoint)
```

Reply:
I can shift it to the view layer which is annotation view. But there will still be chaining in the didPan method because the PanGesture is called for the annotation view.

This is how the parent chaining goes
annotation view --parent--> document view (inherent to PDFKit, we have no class here) --parent--> pdf scrollview (also inherent to PDFKit) --parent--> DocumentPdfView which is a class I have. It inherits PDFView from PDFKit

That's why I think the chaining is inevitable